File: 540. Single Element in a Sorted Array.c
int singleNonDuplicate(int* nums, int numsSize) {
    int i = 0;
    while (i < numsSize - 2){
        if (*(nums + i) !=  *(nums + i + 1)){
            return *(nums + i);
        }else {
            i += 2;
        }
    }
    return *(nums + numsSize - 1);
}

File: 540. Single Element in a Sorted Array.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var singleNonDuplicate = function(nums) {
    for (var i = 0; i <  nums.length - 2; i += 2){
        if (nums[i] != nums[i + 1])
            return nums[i];
    }
    return nums.pop();
};
